src/tracksdata/graph/_sql_graph.py

```diff
@@ -568,8 +568,13 @@ def _update_max_id_per_time(self) -> None:
         have unique IDs.
         """
         with Session(self._engine) as session:
-            for t in session.query(self.Node.t).distinct():
-                self._max_id_per_time[t] = int(session.query(sa.func.max(self.Node.node_id)).scalar())
+            # NOTE: SQLAlchemy returns Row objects when iterating, so we select
+            # scalars to ensure the keys are the raw integer time values.
+            for t in session.scalars(sa.select(self.Node.t).distinct()):
+                max_id = session.scalar(sa.select(sa.func.max(self.Node.node_id)).where(self.Node.t == t))
```

Comment:
Hey @yfukai, I didn't realize at the time, but I think we can remove the loop. Would a `groupby` operation work?

```
stmt = (
    select(self.Node.t, func.max(self.Node.node_id))
    .group_by(self.Node.t)
)
```